### PR TITLE
Make docs site more Membrane-y

### PR DESCRIPTION
**Pull request by juancampa**
Based on the realization that using different font-sizes is what makes the design look different from the main website.

I changed all text to use the same font size. This should also make it look better when embedded in the IDE.

Other changes:
 - Replaces `Membrane-isms` with `Concepts`. I like the term but I felt it was hard to parse and understand what that section was about and I feel like Concepts is pretty standard.
 - Opened all sections by default since we don't have that many pages.
 - Fixed an issue with hovering of code blocks getting clipped.
   Screenshot 2024-11-26 at 14 33 22
 - Fixed link to github repo which was pointing to starlight's repo 😅 
 - Small rewrites here and there.

### Before
Screenshot 2024-11-26 at 14 08 13

### After
Screenshot 2024-11-26 at 14 08 38

### CODE REVIEW

File: src/content/docs/concepts/programs.mdx [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { FileTree } from "@astrojs/starlight/components";
 
 Membrane is similar to an operating system; you deploy programs to the Membrane engine, and it takes care of persistence, communication, logging, resources, etc.
 
-A Membrane _program_ is TypeScript or JavaScript code that runs in Membrane's serverless cloud runtime. Unlike traditional serverless runtimes, it is [stateful](/features/state/) and includes multiple handlers instead of a single function.
+A Membrane _program_ is TypeScript or JavaScript code that runs our cloud runtime. Unlike traditional serverless runtimes, it is [stateful](/features/state/) and includes multiple handlers instead of a single function.
```
Comment (pmillspaugh) [inline src/content/docs/concepts/programs.mdx:9]: "...runs _in_ our cloud runtime."?

Comment (pmillspaugh) [inline src/content/docs/concepts/programs.mdx:9]:
```suggestion
A Membrane _program_ is TypeScript or JavaScript code that runs in our cloud runtime. Unlike traditional serverless runtimes, it is [stateful](/features/state/) and includes multiple handlers instead of a single function.
```

File: astro.config.mjs [hunk 2]
```diff
@@ -103,7 +103,7 @@ export default defineConfig({
       },
       social: {
         discord: "https://discord.gg/4RHyJDV8kj",
-        github: "https://github.com/withastro/starlight",
+        github: "https://github.com/membrane-io/docs",
```
Comment (pmillspaugh) [inline astro.config.mjs:106]: Oof

File: src/content/docs/features/endpoints.md [hunk 3]
```diff
@@ -29,7 +29,7 @@ declare module "membrane" {
 }
 
 // ---cut---
-export const endpoint: resolvers.Root["endpoint"] = (req) => {
+export const endpoint = (req) => {
```
Comment (pmillspaugh) [inline src/content/docs/features/endpoints.md:32]: nice

File: src/styles/base.css [hunk 4]
```diff
@@ -13,6 +13,28 @@ html {
...
+  --sl-text-6xl: 12px;
+}
+
+h1,
+h2,
+h3,
+h4,
+h5,
+h6 {
+  font-size: 12px !important;
```
Comment (pmillspaugh) [inline src/styles/base.css:36]: I do find visual hierarchy of headings helpful on docs pages, especially longer ones. But I'm fine with trying this for now and revisiting later.

File: src/styles/base.css [hunk 5]
```diff
@@ -103,15 +125,12 @@ li > a {
   width: calc(100% - 8px);
   max-width: 720px;
   object-fit: cover;
-  padding: 8px;
+  padding: 20px;
```
Comment (pmillspaugh) [inline src/styles/base.css:128]: What's the motivation for this? It looks like a lot of padding to me now, but I don't feel too strongly

**Review by pmillspaugh — APPROVED** [review]
Nice, thanks for cleaning up!

File: astro.config.mjs (hunk 2 above)
Comment (iamseeley) [reply astro.config.mjs:106]: 🫢

File: src/styles/base.css (hunk 5 above)
Comment (juancampa) [reply src/styles/base.css:128]: Nice catch. Meant to use 10px to keep it consistent with the homepage

**Comment by juancampa** [comment]
My bad. Merged this by accident. Will address feedback in a follow up

File: src/content/docs/features/state.md
Comment (iamseeley) [inline src/content/docs/features/state.md:1]: >Note that the entire JS heap is persistent, not just `state`. So technically, you could use module-level variables to store
state. However, since each update (i.e. code change) creates a new ES Module, prior module-level variables are not accessible from newer modules, hence the need
for a `state` object shared across all versions of the program's code.

This is a little confusing to me. You're saying users could use module-level variables to store state, but they actually can't? 

It does make sense to include a note on "why the state object" like this though.  What do you think @pmillspaugh ?

**Review by iamseeley — COMMENTED** [review]
Looks great! I was late to the party, but I left some notes about the content for `state` and larger headings for better visual hierarchy.

File: src/content/docs/features/state.md
Comment (juancampa) [reply src/content/docs/features/state.md:1]: Yeah I agree. Saying "you could" makes it sound like that's a valid way of doing it.

The reason for the change is that I don't want to bury the fact that the entire heap is persisted. Before, it sounded like `state` is the special object that has persistence, and this confusion has come up a few times during interviews. It makes users think that there's some magic behind it serializing it as JSON.

Open to suggestions on how to to rephrase it but how about this:

> Note that the entire JS heap is persistent, not just `state`. Since each deploy (i.e. code change) creates a new ES Module, this object serves as a convenient way to pass data from one version to the next.

File: src/styles/base.css (hunk 4 above)
Comment (juancampa) [reply src/styles/base.css:36]: Ok yeah you guys are right the hierarchy is lost. I made some changes in #61. Let me know if that improves it.

File: src/content/docs/features/state.md
Comment (pmillspaugh) [reply src/content/docs/features/state.md:1]: Yeah I think that's good. I agree that the language we had before about module-level variables was confusing
